### PR TITLE
sp_exec impl

### PR DESCRIPTION
**Pull request by chrisxu333**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

Currently, this implementation supports following features of stored procedure:
1. variable declaration and usage in any normal sql statements like 'select * from table where id = v1'.
2. dynamic and nested scoping
3. while / repeat / loop structure
4. if-elseif-else / case structure
5. Pass external parameters (system, global, session, user defined) into stored procedure with read or write privileges.

What this implementation does not support:
1. when two loops are nested, it currently cannot jump directly out of the outer loop using label. This will be fixed soon.

### CODE REVIEW

**Review by aressu1985 — APPROVED** [review]

File: pkg/frontend/authenticate.go [hunk 1]
```diff
@@ -1264,6 +1265,8 @@ const (
 	getAccountIdAndStatusFormat = `select account_id,status from mo_catalog.mo_account where account_name = '%s';`
 	getPubInfoForSubFormat      = `select database_name,all_account,account_list from mo_catalog.mo_pubs where pub_name = "%s";`
 	getDbPubCountFormat         = `select count(1) from mo_catalog.mo_pubs where database_name = '%s';`
+
+	fetchSqlOfSpFormat = `select body, args from mo_catalog.mo_stored_procedure where name = '%s' and db = '%s';`
```
Comment (fengttt) [inline pkg/frontend/authenticate.go:1269]: can you use prepare and bind?   Did you checked if this fmt.Sprintf will cause sql injection?

Wait -- I think all the sql above have the same problem.    @daviszhen @ouyuanning ?

**Review by fengttt — COMMENTED** [review]
There is no expected result for the test?

**Comment by fengttt** [comment]
Is this PR still relavent?

**Comment by chrisxu333** [comment]
Yes it's still relevant. I haven't pushed the updated test yet. I was waiting for motester to fix a tiny bug before complete the testing. > Is this PR still relavent?

**Comment by chrisxu333** [comment]
Same thing as commented below. I'll push it soon. > There is no expected result for the test?

**Review by fengttt — APPROVED** [review]
I have reviewed design with Chris, but definitely not deep enough to comment on details.

**Review by ouyuanning — APPROVED** [review]

**Review by daviszhen — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by qingxinhome — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**